sdk/clientcore/core/src/main/java/io/clientcore/core/http/models/RequestOptions.java

```diff
@@ -288,32 +222,18 @@ public RequestOptions setContext(Context context) {
      * @throws IllegalStateException if this instance is obtained by calling {@link RequestOptions#none()}.
      */
     public RequestOptions setLogger(ClientLogger logger) {
-        checkLocked("Cannot set logger.");
         this.logger = logger;
-
         return this;
     }
 
     /**
-     * Locks this {@link RequestOptions} to prevent further modifications.
-     *
-     * @return This {@link RequestOptions} instance.
-     */
-    private RequestOptions lock() {
-        locked = true;
-
-        return this;
-    }
-
-    /**
-     * An empty {@link RequestOptions} that is immutable, used in situations where there is no request-specific
-     * configuration to pass into the request. Modifications to the {@link RequestOptions} will result in an
-     * {@link IllegalStateException}.
+     * An empty {@link RequestOptions} used in situations where there is no request-specific
```

Comment:
Should call out somewhere in this documentation that if any of the setter methods are called when the `RequestOptions == none()` that it will throw an exception.